libbpf-cargo/src/build.rs

```diff
@@ -179,7 +179,7 @@ fn compile_one(
 
     if !clang_args
         .iter()
-        .any(|arg| arg.as_os_str() == OsStr::new("-D__TARGET_ARCH_"))
+        .any(|arg| arg.to_string_lossy().contains("__TARGET_ARCH_"))
```

Comment:
you actually need to check if it's a prefix of a string, not really "contains"

Reply:
Both `-DXXX` and `-D XXX` are valid argument specifications, from what I understand. So I don't think we can assume that it's a strict prefix.

Reply:
ah, the original sting contains `-D`. It's probably fine in practice to just go with contains, unlikely someone will do `-Dhaha__TARGET_ARCH_blah`, but technically speaking you are matching more stuff than you actually intend to.